apis/projectcontour/v1alpha1/contourconfig.go

```diff
@@ -289,6 +289,10 @@ type EnvoyListenerConfig struct {
 	// See: https://github.com/projectcontour/contour/issues/3221
 	DisableAllowChunkedLength bool `json:"disableAllowChunkedLength"`
 
+	// MergeSlashes toggles Envoy's non-standard merge_slashes path transformation option
+	// which strips duplicate slashes from request URL paths.
+	MergeSlashes bool `json:"mergeSlashes"`
```

Comment:
I think this should be switched to `DisableMergeSlashes`, following a similar pattern as for the above field `DisableAllowChunkedLength`, with a default value of `false`. This way, Contour's default behavior doesn't change, and the user only needs to specify something if they want to change the default behavior. Otherwise, we run into problems where we can't distinguish between this field not being specified, and being explicitly specified as `false`.

Reply:
Thanks, this sounds good. I thought applying the `kubebuilder` default would have sufficed for that but I see now that it only controls the generated example resources.